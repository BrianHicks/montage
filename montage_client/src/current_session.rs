#[cynic::schema("montage")]
mod schema {}

#[derive(cynic::QueryFragment, Debug)]
#[cynic(graphql_type = "Query")]
pub struct CurrentSessionQuery {
    pub current_session: Option<Session>,
}

#[derive(cynic::QueryFragment, Debug)]
pub struct Session {
    pub description: String,
    pub duration: Duration,
    pub end_time: Option<DateTime>,
    pub kind: Kind,
    pub projected_end_time: DateTime,
    pub remaining_time: Option<Duration>,
    pub start_time: DateTime,
}

#[derive(cynic::Enum, Clone, Copy, Debug)]
pub enum Kind {
    Task,
    Break,
}

type DateTime = chrono::DateTime<chrono::Local>;
cynic::impl_scalar!(DateTime, schema::DateTime);

type Duration = iso8601::Duration;
cynic::impl_scalar!(Duration, schema::Duration);

#[cfg(test)]
mod test {
    use super::*;
    use cynic::QueryBuilder;

    #[test]
    fn current_session_gql_output() {
        let operation = CurrentSessionQuery::build(());

        insta::assert_snapshot!(operation.query);
    }
}
