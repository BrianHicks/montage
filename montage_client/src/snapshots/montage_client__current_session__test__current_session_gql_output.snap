---
source: montage_client/src/current_session.rs
expression: operation.query
---
query CurrentSessionQuery {
  currentSession {
    description
    duration
    endTime
    kind
    projectedEndTime
    remainingTime
    startTime
  }
}


